src/main/java/gregtech/common/blocks/MetaBlocks.java

```diff
@@ -106,36 +107,40 @@ public static void init() {
 
         COMPRESSED = new HashMap<>();
         ORES = new HashMap<>();
-        ArrayList<DustMaterial> materialBuffer = new ArrayList<>();
+        Material[] materialBuffer = new Material[16];
+        Arrays.fill(materialBuffer, Materials._NULL);
         int generationIndex = 0;
         for(Material material : Material.MATERIAL_REGISTRY.getObjectsWithIds()) {
             if(material instanceof DustMaterial) {
-                materialBuffer.add((DustMaterial) material);
-                if(materialBuffer.size() == 16) {
-                    createCompressedBlock(materialBuffer, generationIndex);
-                    materialBuffer.clear();
-                    generationIndex++;
-                }
+            	int id = Material.MATERIAL_REGISTRY.getIDForObject(material), index = id / 16;
```

Comment:
I spend some time looking where index was declared. Separate this line into two and we are good to merge